### ISSUE TITLE
YESTERDAY'S RELEASE : Lot of error when build AOT

### ISSUE THREAD

**Issue by istiti** [angular/angular-cli#5620]
> After updating (existing project, on windows10)  :
> CLI from rc2 to v1.0.0
> Angular from rc4 to v4.0.0
> Typescript to 2.2.1

Before yesterday's release, I never get errors when building using `--prod` but today my projects doesn't compile cause of private/protected members used in template!!! WHAT A MESS 

https://gist.github.com/istiti/5b8e8c9db15135681f605b98e5a8e4f1

Wonder why this never happends before? how can I now search/replace all private/protected members used in template to public ! 

it's maybe stupid enhancement but is it possible that cli convert all members to public  when runnin ng build ? or add a command to overwrite these propertys to public directly in project?

N.B: relativly big project

cc @hansl  @filipesilva @Brocco @IgorMinar

**Comment by alastair-todd** [comment]
+1 could someone please explain this behaviour?

1) for html templates to access their parent component's properties, what accessor level *should* these properties be (private is the gut feeling, the html + component class make up the component, I shouldnt have to make them public)?
2) why would the --prod flag change this behaviour?

**Comment by bogdvn** [comment]
+1
Beside this errors:

[disabled]="OnDsNotSelected()"
error: Supplied parameters do not match any signature of call target.

 OnDsNotSelected(): boolean {
        return this.options.length == 0;
    }

Why i get this  ?

**Comment by stavn1** [comment]
Is there a way to solve this error with angular 2.4?

**Comment by Thieus** [comment]
+1  Same error
Have you got a solution ?

```
ERROR in ....ngfactory.ts (696,41): Property 'clipPathId' does not exist on type 'BubbleChartComponent'.
ERROR in ....ngfactory.ts (207,35): Supplied parameters do not match any signature of call target.
ERROR in ....ngfactory.ts (642,48): Property 'onActivate' does not exist on type 'HeatMapComponent'.
ERROR in ....ngfactory.ts (646,48): Property 'onDeactivate' does not exist on type 'HeatMapComponent'.
ERROR in ....ngfactory.ts(342,44): Property 'labels' does not exist on type 'AdvancedPieChartComponent'.
ERROR in ....ngfactory.ts (300,67): Supplied parameters do not match any signature of call target.
ERROR in ....ngfactory.ts (131,7): Supplied parameters do not match any signature of call target.
ERROR in ....ngfactory.ts (870,44): Property '_selectedType' is private and only accessible within class 'SearchFormFlowDetailComponent'.
....ngfactory.ts (879,44): Property '_typesFlow' is private and only accessible within class 'SearchFormFlowDetailComponent'.
```

**Comment by hassanasad** [comment]
Could all these errors be related to TypeScript version 2.2.1 ?

**Comment by filipesilva** [comment]
User opened issue without bothering to fill in issue template, no repro or anything!!! WHAT A MESS

**Comment by IgorMinar** [comment]
@filipesilva please be nice. Your comment is in violation of our [communication guidelines](http://goo.gl/p7BoeV).

@istiti @Thieus @alastair-todd @bogdvn @hassanasad can one of you please open a new issue following the instructions in the issue template? We need all the info we can get to reproduce and confirm this issue before we can quickly fix it.

**Comment by istiti** [comment]
Thanks to this guy will open new issue sorry for miss template issue

Ps: i know for AoT all members need to be public (read this more than 5months !!) and how class and template are connected but i never get this error before and CLI didn't write on readme that this is not checked and will be in release  ! I have no time to check in the past in present in furthur all articles blog on web !!! Need to be productive!! I assum this is not how open source work so please apologize WHAT A MESS my project  is big and i dont want search/replace whole my app members to public to lost objectoriented principes 

Please apologize

**Comment by alastair-todd** [comment]
Declare a private variable in a Component and access it in the html. 

Then ng build with the --prod flag.

Nothing complicated but fairly major.

**Comment by sumitarora** [comment]
Issue is being tracked on this one https://github.com/angular/angular-cli/issues/5623
☝️ ☝️ ☝️

**Comment by clydin** [comment]
Please see this issue for additional information: https://github.com/angular/angular-cli/issues/5623

**Comment by netmikey** [comment]
If someone is looking to migrate a bigger codebase: I've ~~created~~ hacked together a small macOS script that patches your codebase to make the required fields and methods (those actually used in at least one template) public. Currently only tested (and probably only works) on macOS because of sed vs GNU-sed specifics.

https://gist.github.com/netmikey/cb29ac391953f2c67aa102b28fd4fe16

**Comment by pcgeng** [comment]
I have resolved it with Angular **2.4.10**, Angular-CLI **1.0.0-rc.2,** and TypeScript **2.1.6**

**Comment by wnabil** [comment]
+1 for 1.0.0 but i dont have it on 1.0.0-beta.23.3

**Comment by kblestarge** [comment]
@netmikey your script sounds like it has the potential to save me a lot of time, but it doesn't seem to be working for me.

Are there any other workarounds for this issue?

**Comment by oferns** [comment]
I am getting this error with @angular/cli v1.3.1 / Typescript 2.4.2 / @angular/core v4.3.5 ng build works. ng build --prod throws lots of errors like
 ERROR : Property 'providers' is private and only accessible
 within class 'ProviderPickerComponent'.

Do we need to reopen this issue?

**Comment by wnabil** [comment]
@oferns There is a changes was made with the angular cli from while ago.
the main problem is some variables are used inside the html and they are private in the component class
just remove the private key and everything will work fine.
I know this may be a huge problem specially with a big project "i felt that pain actually with my project xD" but sadly there is no other way to fix this.

**Comment by oferns** [comment]
@wnabil Thanks for the response. Basically, dont make private anything that the view/html needs to see? How come it doesnt fail in dev mode? thanks

**Comment by wnabil** [comment]
@oferns as i understand before they was taking about this error will appear only when building the aot or the prod. generally just keep your variables public if you are using them inside the html.
you can serve your application with angular-cli as prod and aot "for testing and to help you with fixing the errors only" by running:
`ng serve --prod --aot`.
